spec/features/date_selection_spec.rb

```diff
@@ -0,0 +1,195 @@
+RSpec.describe 'date selection', type: :feature do
+  include GdsApi::TestHelpers::ContentDataApi
+
+  before do
+    initial_page_stub
+  end
+
+  it 'renders data for the last 30 days if no date period is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 30.days
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+
+    expect_default_date_period_metrics_to_be_displayed(from, to)
+  end
+
+  it 'renders data for the last 30 days when `last 30 days` is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 30.days
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+    find('#date_range_last-30-days').click
+    click_button 'OK'
+    sleep 2
```

Comment:
I would get rid of the `sleep 2` by using an assertion:

```ruby
expect(page).to have_selector('selector to the area where the click button lives')
```

This way we will be waiting for the button to appear, and as a side effect, we are also sure that the section exists.

Reply:
The sleeps are not necessary, forgot to remove a couple sorry! 🤦‍♂️